lib/Logger.ts

```diff
@@ -50,15 +50,26 @@ class Logger {
 
   private level: string;
   private context: Context;
+  private subcontext?: string;
   private logger?: winston.Logger;
+  private filename?: string;
   private instanceId: number;
-
-  constructor({ level, filename, context, instanceId, disabled, dateFormat }:
-    {instanceId?: number, level?: string, filename?: string, context?: Context, disabled?: boolean, dateFormat?: string}) {
-
-    this.level = level || Level.Trace;
-    this.context = context || Context.Global;
-    this.instanceId = instanceId || 0;
+  private dateFormat?: string;
+
+  constructor({ level = Level.Trace, filename, context = Context.Global, subcontext, instanceId = 0, disabled, dateFormat }: {
```

Comment:
minor style nit: I feel like this line is too long.

Reply:
I was on the fence here too, so I broke it down to one variable per line.